File: src/android/wrapper/LocationTrackingConfig.java
package edu.berkeley.eecs.emission.cordova.tracker.wrapper;

import com.google.android.gms.location.LocationRequest;
import edu.berkeley.eecs.emission.cordova.tracker.Constants;

/**
 * Created by shankari on 10/20/15.
 */

public class LocationTrackingConfig {

    private static final int FIVE_MINUTES_IN_SEC = 5 * 60;

    private boolean is_duty_cycling = true;
    private boolean simulate_user_interaction = false;
    private int accuracy = LocationRequest.PRIORITY_HIGH_ACCURACY;
    private int accuracy_threshold = 200;
    private int filter_distance = -1; // unused
    private int filter_time = Constants.THIRTY_SECONDS;
    private int geofence_radius = Constants.TRIP_EDGE_THRESHOLD;
    private int trip_end_stationary_mins = 5;
    private int android_geofence_responsiveness = 5 * Constants.MILLISECONDS;
    private boolean is_fleet = false;

    // ios fields not set
    private boolean ios_use_visit_notifications_for_detection;
    private boolean ios_use_remote_push_for_sync;

    public boolean isDutyCycling() {
        return this.is_duty_cycling;
    }

    public boolean isSimulateUserInteraction() {
        return this.simulate_user_interaction;
    }

    public int getAccuracy() {
        return this.accuracy;
    }

    public int getAccuracyThreshold() {
        return this.accuracy_threshold;
    }

    public int getFilterDistance() {
        return this.filter_distance;
    }

    public int getFilterTime() {
        return this.filter_time;
    }

    public int getGeofenceRadius() {
        return this.geofence_radius;
    }

    public int getTripEndStationaryMins() {
        return this.trip_end_stationary_mins;
    }

    public int getResponsiveness() {
        return this.android_geofence_responsiveness;
    }

    public boolean isFleet() {
        return this.is_fleet;
    }

    // We don't need any "set" fields because the entire document will be set as a whole
    // using the javascript interface
}

File: src/ios/Wrapper/LocationTrackingConfig.m
//
//  LocationTrackingConfig.m
//  CFC_Tracker
//
//  Created by Kalyanaraman Shankari on 11/6/15.
//  Copyright © 2015 Kalyanaraman Shankari. All rights reserved.
//

#import "LocationTrackingConfig.h"
#import "BEMBuiltinUserCache.h"
#import <CoreLocation/CoreLocation.h>

#define kOneMeter 1 // in meters
#define kFiveMeters 5 // in meters
#define kFifty_Meters 50 // in meters
#define kHundred_Meters 100 // in meters
#define kTripEndStationaryMins 10 // in minutes

@implementation LocationTrackingConfig

-(id)init {
    self.is_duty_cycling = YES;
    self.simulate_user_interaction = NO;
    self.accuracy = kCLLocationAccuracyBest;
    self.accuracy_threshold = 200;
    self.filter_distance = kOneMeter;
    self.filter_time = -1; // unused
    self.geofence_radius = kHundred_Meters;
    self.trip_end_stationary_mins = kTripEndStationaryMins;
    self.ios_use_visit_notifications_for_detection = YES;
    self.ios_use_remote_push_for_sync = YES;
    self.android_geofence_responsiveness = -1; // unused
    return self;
}

@end
